pkg/operator/ceph/file/filesystem.go

```diff
@@ -157,6 +165,21 @@ func validateFilesystem(context *clusterd.Context, clusterInfo *cephclient.Clust
 	return nil
 }
 
+func hasDuplicatePoolNames(poolSpecList []cephv1.NamedPoolSpec) bool {
+	poolNames := make(map[string]struct{})
```

Comment:
`poolNames := make(map[string]int)`

Reply:
Not storing anything as value here. So `struct{}` should be good. Its 0-sized so no memory is required to store `struct{}`

Reply:
int will have a default value of `0`

Reply:
Ohkay, make sense:)

Reply:
`map[Type]struct{}` is a low-overhead way of approximating a set data structure in Golang.